### ISSUE TITLE
exclusive locks not supported on vagrant box nfs

### ISSUE THREAD

**Issue by cordoval** [thephpleague/flysystem#445]
I am getting this error when inside vagrant. It works when I remove the LOCK_EX, any ideas?

```php
[Bldr\Exception\TaskRuntimeException] 
There was an error running the exec task. Output: 


[Symfony\Component\Debug\Exception\ContextErrorException 
] 
Warning: file_put_contents(): Exclusive locks are not su 
pported for this stream 

```

**Comment by frankdejonge** [comment]
I've setup nfs version 3 for my shared folder:

```
config.vm.synced_folder "./", "/vagrant", type: "nfs", mount_options: ['rw', 'vers=3', 'tcp', 'fsc']
```

Does that work for you?

**Comment by cordoval** [comment]
thanks for the quick response @frankdejonge :older_man: I will try when at home :airplane:

**Comment by frankdejonge** [comment]
@cordoval I reckon this worked. If not, feel free to comment and I'll reopen the issue.

**Comment by cordoval** [comment]
i had to use the different method, all good, thanks and apologize for not closing earlier.

**Comment by frankdejonge** [comment]
No worries, I know how these things go :) thanks for the verification.

**Comment by yohendry** [comment]
@cordoval can u provide your solution?

**Comment by approached** [comment]
+1
@cordoval can u provide your solution?
...same problem

**Comment by cordoval** [comment]
I don't remember, sorry

**Comment by Aeon** [comment]
DAMNIT!

![XKCD: Wisdom of the Ancients](https://imgs.xkcd.com/comics/wisdom_of_the_ancients.png)

**Comment by approached** [comment]
@yohendry this is the solution:
```
config.vm.synced_folder ".", "/vagrant", id: "v-root", mount_options: ["rw", "tcp", "nolock", "noacl", "async"], type: "nfs", nfs_udp: false
```

**Comment by twify93** [comment]
@approached Apparently for Linux users this works, but for Windows users - it doesn't...

**Comment by jwhulette** [comment]
For FreeBSD:
  # Place in /etc/rc.conf, to resolve error: exclusive locks not supported
  rpc_lockd_enable="YES"
  rpc_statd_enable="YES"

**Comment by chrisandrewcl** [comment]
Just in case someone lands here from Google: I lost some considerable time trying to figure this out. I have tried several different configurations. Nothing worked, so I stepped down a level and tried to configure it directly between the guest instance and the host machine (i.e. manually editing `/etc/exports` and issuing `mount`s). It paid off :)

The final configuration that worked for my setup (Ubuntu 17.10, Vagrant 2.0.2) is this:

``` 
config.vm.synced_folder ".", "/vagrant", 
    :nfs => true, 
    :nfs_version => 4, 
    :nfs_udp => false, 
    :mount_options => ['sec=sys']
```
